### PR TITLE
Document domctl.h

### PR DESCRIPTION
**Pull request by Deedone**
Document all of the public functions in the domctl API with oxygen
Document all of the public functions in the sysctl API with doxygen
Fixes https://github.com/xen-troops/zephyr/issues/97

### CODE REVIEW

File: include/zephyr/xen/dom0/domctl.h [hunk 1]
```diff
@@ -12,27 +12,190 @@
```
Comment (GrygiriiS) [inline include/zephyr/xen/dom0/domctl.h:19]: You probably wanna add

-  file header, like 
```
/**
 * @file
 *
 * @brief Public APIs for the I2C drivers.
 */
```
- interface header, like
```
/**
 * @brief I2C Interface
 * @defgroup i2c_interface I2C Interface
 * @ingroup io_interfaces
 * @{
 */
```

- interface footer
```
/**
 * @}
 */
```

File: include/zephyr/xen/dom0/domctl.h [hunk 2]
```diff
@@ -12,27 +12,190 @@
 
 #include <zephyr/kernel.h>
 
+/**
+ * @brief Perform a scheduler operation on a specified domain.
+ *
+ * @param domid The ID of the domain on which the scheduler operation is to be performed.
+ * @param[in,out] sched_op A pointer to a `struct xen_domctl_scheduler_op` object that defines
```
Comment (GrygiriiS) [inline include/zephyr/xen/dom0/domctl.h:26]: Have you tried '@ref xen_domctl_scheduler_op' here? 

Actually `refs` might be used in all APIs docs to reference Xen structures.

File: include/zephyr/xen/dom0/domctl.h [hunk 3]
```diff
@@ -12,27 +12,190 @@
...
+ *                         the specific scheduler operation to be performed.
+ * @return Returns 0 on success, or a negative error code on failure.
+ */
 int xen_domctl_scheduler_op(int domid, struct xen_domctl_scheduler_op *sched_op);
+/**
```
Comment (GrygiriiS) [inline include/zephyr/xen/dom0/domctl.h:32]: Add blank lines before each API

File: include/zephyr/xen/dom0/domctl.h [hunk 4]
```diff
@@ -12,27 +12,190 @@
...
+ */
+int xen_domctl_set_paging_mempool_size(int domid, uint64_t size);
+/**
+ * @brief Sets the maximum memory for a specified domain.
+ *
+ * @param domid The domain ID of the domain to set the maximum memory for.
+ * @param max_memkb The maximum memory (in kilobytes) to set for the domain.
+ * @return 0 on success, or a negative error code on failure.
+ */
 int xen_domctl_max_mem(int domid, uint64_t max_memkb);
```
Comment (GrygiriiS) [inline include/zephyr/xen/dom0/domctl.h:106]: TODO: this api doesn't follow `set/get`  naming convention

Comment (Deedone) [reply include/zephyr/xen/dom0/domctl.h:106]: Function names are chosen based on defines from Xen headers.

File: include/zephyr/xen/dom0/domctl.h (hunk 2 above)
Comment (Deedone) [reply include/zephyr/xen/dom0/domctl.h:26]: Xen public headers do not use doxygen so there is nothing to ref to.

File: include/zephyr/xen/dom0/domctl.h [hunk 5]
```diff
@@ -12,27 +12,214 @@
 
 #include <zephyr/kernel.h>
 
+/**
+ * @file
+ *
+ * @brief Xen Domain Control Interface
+ */
+
```
Comment (GrygiriiS) [inline include/zephyr/xen/dom0/domctl.h]: if you check other generic headers - "file header" is added as first block in the file or right after copyright block, can we follow the same, pls.

**Comment by GrygiriiS** [comment]
It seems to finish this task properly - the xen_xx.rst file need to be added in doc folder.

It has to be similar to doc/hardware/peripherals/spi.rst (this one very simple) or doc/hardware/peripherals/i2c.rst (or any other interface doc).

Question where to add these Xen doc files, with assumption that more docs will be added (event, ranges, gnt) - is "doc/hardware/peripherals" proper path? or need Xen specific folder somewhere?

**Comment by Deedone** [comment]
Let's try to keep PRs small and finally merge something because we keep getting interrelated changes piling up, and this is getting out of hand.

**Review by firscity — APPROVED** [review]
`Acked-by: Dmytro Firsov <dmytro_firsov@epam.com>`

**Comment by GrygiriiS** [comment]
LGTM. Thank you very much.

**Pull request merged**